lib/live_debugger/app/debugger/node_state/web/hook_components/assigns_display.ex

```diff
@@ -0,0 +1,100 @@
+defmodule LiveDebugger.App.Debugger.NodeState.Web.HookComponents.AssignsDisplay do
+  @moduledoc """
+  LiveComponent that can be used to display a tree of terms.
+  It removes children of collapsed nodes from HTML, and adds them when the node is opened.
+  """
+
+  use LiveDebugger.App.Web, :hook_component
+
+  alias LiveDebugger.App.Debugger.Web.Components.ElixirDisplay
+  alias LiveDebugger.App.Utils.TermNode
+  alias LiveDebugger.App.Utils.TermParser
+  alias Phoenix.LiveView.AsyncResult
+
+  @required_assigns [:node_assigns_info]
+
+  @impl true
+  def init(socket) do
+    socket
+    |> check_assigns!(@required_assigns)
+    |> attach_hook(:assigns_display, :handle_event, &handle_event/3)
+    |> register_hook(:assigns_display)
+  end
+
+  attr(:id, :string, required: true)
+  attr(:node, TermNode, required: true)
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div id={@id}>
+      <.term node={@node} />
```

Comment:
What's the point of having the whole HTML structure of this component extracted to the private function?